lib/embed/embed.dart

```diff
@@ -337,12 +337,43 @@ class PlaygroundMobile {
     }
   }
 
+  bool validFlex(String input) {
+    return input != null && double.parse(input) > 0.0 && double.parse(input) < 1.0;
+  }
+  
+  int roundFlex(double flex) {
```

Comment:
Nit: you'll want spaces around the operator, and this method is short enough that you should prefer the function shorthand here:

```
int roundFlex(double flex) => (flex * 10.0).round();
```